tests/shared/src/test/scala/scala/meta/tests/parsers/LitSuite.scala

```diff
@@ -5,93 +5,104 @@ import scala.meta._
 import scala.meta.dialects.Scala211
 
 class LitSuite extends ParseSuite {
+
+  // Structure does not always return consisten results
+  private def assertLiteral(obtained: Term, expected: Lit) {
+    obtained match {
+      case lit: Lit => assertEquals(lit.value, expected.value)
+      case _ => fail(s"Expected literal, got ${obtained.structure}")
+    }
+  }
+
   test("true") {
-    val Lit(true) = term("true")
+    assertTree(term("true"))(Lit.Boolean(true))
   }
 
   test("false") {
-    val Lit(false) = term("false")
+    assertTree(term("false"))(Lit.Boolean(false))
   }
 
   test("42") {
-    val Lit(42) = term("42")
+    assertTree(term("42"))(Lit.Int(42))
   }
 
   test("2147483648") {
     intercept[ParseException] { term("2147483648") }
   }
 
   test("2147483647") {
-    val Lit(2147483647) = term("2147483647")
+    assertTree(term("2147483647"))(Lit.Int(2147483647))
   }
 
   test("-2147483648") {
-    val Lit(-2147483648) = term("-2147483648")
+    assertTree(term("-2147483648"))(Lit.Int(-2147483648))
```

Comment:
this works...

Reply:
It works if printing via `.structure` is the same. But if it prints differently yet it's a correct value it will be reported as an error.

Reply:
Previously, we would do an unapply which only compares values, so that would be correct. When using `Lit.Int(0xCAFEBABE)` the int inside will be printed normally, so we would get a mismatch.